### PR TITLE
Closed state

### LINKED ISSUE #286: Closed state name collection
When the app is closed, have a form to INFORM me more. 
aka, alway be collecting (user info)

-------

__Update:__ Things to do –

 * [x] Add `show_candidates` setting to show or hide candidate listing.
 * [x] Add endpoint for closed state form.
 * [x] Put users into a transactional bucket when they submit the closed state form.

:computer:

### PR DESCRIPTION
**Pull request by DFurnes**
# Changes

Closes #286. This hooks up the closed state form to forward users to Message Broker.

It also creates a wrapper [`MessageBroker`](https://github.com/DFurnes/voting-app/blob/closed-state/app/Services/MessageBroker.php) service class that handles configuring the `AMQPConnection` and adding common configuration values (like `application_id`, `mailchimp_grouping_id`, `mc_opt_in_path_id`, etc.) so we don't have to duplicate that logic for every single request.

For review: @angaither @DeeZone

### CODE REVIEW

File: database/migrations/2015_07_27_155237_make_northstar_id_nullable.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class MakeNorthstarIdNullable extends Migration {
```
Comment (angaither) [inline database/migrations/2015_07_27_155237_make_northstar_id_nullable.php:6]: who knew we were storing this in the voting app!? haha, missed that!

**Comment by angaither** [comment]
looks good from this end :+1:

File: database/migrations/2015_07_27_155237_make_northstar_id_nullable.php (hunk 1 above)
Comment (DFurnes) [inline database/migrations/2015_07_27_155237_make_northstar_id_nullable.php:6]: :dizzy:

File: app/Handlers/Events/SendFirstVoteMessage.php [hunk 2]
```diff
@@ -73,8 +62,8 @@ public function handle(UserCastFirstVote $event)
             ];
         }
 
-        $payload = serialize($payload);
-        $broker->publishMessage($payload);
+        $routingKey = env('VOTE_ROUTING_KEY', 'votingapp.event.vote');
+        $this->broker->publish('vote', $payload, $routingKey);
```
Comment (deezone) [inline app/Handlers/Events/SendFirstVoteMessage.php:66]: Ha, you beat me to it! I'm in the process of updating `publishMessage()` method in Message Broker to use the routingKey value as a parm.

Comment (DFurnes) [inline app/Handlers/Events/SendFirstVoteMessage.php:66]: Nice! Looking forward to that update. :smile:

File: app/Http/Controllers/UsersController.php [hunk 3]
```diff
@@ -45,10 +47,41 @@ public function index()
...
+            'country_code' => $user->country_code,
+        ];
 
+        // Send fields for domestic users
+        if($user->country_code === 'US') {
+            $payload['mobile'] = $user->phone;
+            $payload['mobile_tags'] = [
+                env('APP_NAME_TAG', 'votingapp'),
+            ];
+        }
```
Comment (deezone) [inline app/Http/Controllers/UsersController.php:66]: Why not `} else {` here?

Comment (DFurnes) [inline app/Http/Controllers/UsersController.php:66]: I feel like it makes each block a bit more readable to treat them as separate "paragraphs", but it'd work just as well with an `} else {`.

**Comment by deezone** [comment]
Any chance of getting access to the Wercker results? I tried taking a look but my account "DeeZone" (via github) was not allowed access. I'd like to revisit how you set it up and the kinds of results you can get.

Just looking over your shoulder.

**Comment by deezone** [comment]
:+1:

**Comment by DFurnes** [comment]
@DeeZone Sure! Just added you to the [dosomething](https://app.wercker.com/#dosomething) org so you should be able to see everything now!

**Comment by deezone** [comment]
Also note there's a classes in development (to be released this week) in the MB_Toolbox library to handle configuration of the AMQPConnection: https://github.com/DoSomething/mb-toolbox/tree/master/src. I'll poke you for your review.

**Comment by DFurnes** [comment]
@DeeZone Awesome!

**Pull request merged**